### ISSUE TITLE
React hooks

### ISSUE THREAD

**Issue by MartinN3** [lundegaard/react-union#327]
Hello guys,

Are you planning on supporting hooks? In current version i bumped into issue where you can't run production build due to uglify plugin.

I wanted to override/extend webpack config but i dont think i can override uglifyjs, or am i wrong? Thanks

**Comment by aizerin** [comment]
Hi,
It seems that this is a general problem with hooks. I will look at it, but it should be possible to override config.

**Comment by aizerin** [comment]
Do you have a minimal working example? I tried this code with latest react-union and it seems ok.

```js
import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const Root = ({ namespace, data }) => {
	const [state, setState] = useState('defaultState');
	useEffect(() => {
		setState('one time effect');
	}, []);
	return (
		<div>
			{state}
			<button onClick={() => setState('click state')} />
		</div>
	);
};
```

**Comment by MartinN3** [comment]
mmm, seems like it should have crashed on your example - ill try my self again.

Meanwhile if you wanted to know more about this issue community has with hooks and uglifyJS - 
[](https://github.com/facebook/react/issues/14014)

Also back to **dirty solution**. Is there a possibility to override uglifyJs in react-union-scripts without going straight into node_modules? 

`const prodWebpack = () =>
		merge(
			commonConfig,
			{
				bail: true,
				devtool: sourceMaps ? 'source-map' : false,
			},
			//uglifyJsPlugin(cli.verbose, config)
		);` 

Thanks

**Comment by tommmyy** [comment]
@MartinN3 issue with uglify is resolved in react-union-scripts@0.13.1
